src/timedata.cpp

```diff
@@ -98,11 +99,12 @@ void AddTimeData(const CNetAddr& ip, int64_t nOffsetSample)
         }
 
         if (LogAcceptCategory(BCLog::NET)) {
+            std::string logMessage{"time data samples: "};
             for (const int64_t n : vSorted) {
-                LogPrint(BCLog::NET, "%+d  ", n); /* Continued */
+                logMessage += strprintf("%+d  ", n);
```

Comment:
Note that concatenating to a std::string can be inefficient, since it may need to reallocate multiple times if the buffer isn't large enough. You could use str::string::reserve to avoid reallocations, but I don't think it's worth it here, since this log is built only a few times during each run of the application.

Reply:
I agree - will keep this in mind, thanks!

Reply:
tinyformat also isn't really optimized for speed, and using it this way it returns a temporary string, which is copied then deallocated, every time it's called

but yeah it's for optional debug logging, not building a 500MB XML document, i don't think performance at this level is any concern here